src/components/Navigation.vue

```diff
@@ -60,7 +59,11 @@ export default {
 	methods: {
 		logOut: function () {
 			this.$store.dispatch('logout')
-		}
+		},
+		navigateToSignin() {
+			this.$store.commit('setRedirectPath', this.$route.path);
+			VueRouter.push('/signin');
```

Comment:
I believe you meant to handle this with the dispatched event. There is no need to import vuerouter into this component, the router is already available for use everywhere. Remove this line and the import statement above. Inside your setRedirectPath action, use a mutation to redirect user to the signin page